### PR TITLE
add RBD Mirror monitoring alerts

### PR DESCRIPTION
**Pull request by aruniiird**
This is a backport of PR: https://github.com/ceph/ceph/pull/50132 to reef branch, adding RBD Mirror monitoring alerts

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: monitoring/ceph-mixin/prometheus_alerts.yml [hunk 1]
```diff
@@ -714,3 +714,45 @@ groups:
...
+        annotations:
+          description: "Number of image replications per daemon is not suppossed to go beyond threshold 100"
+          summary: "Number of image replications are now above 100"
+        expr: "sum by (ceph_daemon, namespace) (ceph_rbd_mirror_snapshot_image_snapshots) > 100"
+        for: "1m"
+        labels:
+          oid: "1.3.6.1.4.1.50495.1.2.1.10.2"
+          severity: "critical"
+          type: "ceph_default"
+      - alert: "CephRBDMirrorImagesNotInSync"
```
Comment (sunilangadi2) [inline monitoring/ceph-mixin/prometheus_alerts.yml:729]: For how many images does this alert will be triggered?

**Comment by cloudbehl** [comment]
@aruniiird Thanks for PR. Backport guidelines needs followed.

**Review by avanthakkar — APPROVED** [review]

**Comment by aruniiird** [comment]
> @aruniiird Thanks for PR. Backport guidelines needs followed. https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst
> 
> Looks like main PR doesn't follow as well: #50132 https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst

Thanks @cloudbehl , will take a look...

File: monitoring/ceph-mixin/prometheus_alerts.yml (hunk 1 above)
Comment (aruniiird) [reply monitoring/ceph-mixin/prometheus_alerts.yml:729]: AFAI understand, this should be triggered even for a single image miss-match (where local vs remote images are not in sync)

**Review by jmolmo — APPROVED** [review]

**Pull request merged**